app/javascript/src/ActiveDocs/OAS3Autocomplete.ts

```diff
@@ -120,3 +120,15 @@ export const autocompleteOAS3 = async (response: SwaggerUIResponse, accountDataU
     text: JSON.stringify(body)
   }
 }
+export const autocompleteInterceptor = async (response: SwaggerUIResponse, specUrl: string | undefined, accountDataUrl: string, serviceEndpoint: string): Promise<Response> => {
+  // only process the response for Open API spec requests (not the actual API calls responses)
+  let interceptedResponse = response as Response
+  if (specUrl === response.url) {
+    try {
+      interceptedResponse = await autocompleteOAS3(response, accountDataUrl, serviceEndpoint)
+    } catch (error: unknown) {
+      console.error(error)
+    }
+  }
+  return interceptedResponse
```

Comment:
This function is a bit messy and that's partly because of the lack of types. The return type of responseInterceptor is `Promise<SwaggerUIResponse> | SwaggerUIResponse` so there's no need to make the function async or parse the return value into `Response`.

`autocompleteOAS3` also takes care of any possible async errors so no need for a try/catch block. I suggest the folllowing:
```ts
export const autocompleteInterceptor = (response: SwaggerUIResponse, accountDataUrl: string, serviceEndpoint: string, specUrl?: string): Promise<Response> | SwaggerUIResponse => {
  // Don't process actual API calls responses
  if (specUrl !== response.url) {
    return response
  }

  return autocompleteOAS3(response, accountDataUrl, serviceEndpoint)
}
```

Reply:
Thanks for that! Much better :)

I dunno, I just entered in a loop where any thing I tried was giving some warning, "await in the this context is not allowed" and some others... I tried to avoid these errors, and yes, the outcome was ugly. I've applied your suggestions.